tests/testutils/goroutineassistant.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutils
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+)
+
+type GoroutineAssistant struct {
+	wait chan bool
+	fail chan bool
+	l    sync.Mutex
+	wg   sync.WaitGroup
+	t    *testing.T
+	s    string
+}
```

Comment:
I think you can implement this with one channel and a wait group:

```
type GoroutineAssistant struct {
    fail chan string
    wg   sync.WaitGroup
    t    *testing.T
}
```
And then combine Wait and Finish:
```
func (a *GoroutineAssistant) Wait() {
    a.wg.Wait()
    select {
        case s := <- a.fail:
            t.Fatalf(s)
        default:
    }
}
```